### PR TITLE
[improvement] createManifest no longer requires compiling project source code

### PR DESCRIPTION
**Pull request by ferozco**
## Before this PR
To discover in repo product dependencies, `createManifest` would cause all project dependencies to run `Jar` causing them to fully compile.

## After this PR
Project dependencies are no longer compiled

### CODE REVIEW

File: gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/CreateManifestTask.java [hunk 1]
```diff
@@ -127,9 +133,17 @@ final void setManifestExtensions(Map<String, Object> manifestExtensions) {
         return ignoredProductIds;
     }
 
-    @InputFiles
-    final Property<Configuration> getProductDependenciesConfig() {
-        return productDependenciesConfig;
+    @Input
+    final Set<String> getProductDependenciesConfig() {
```
Comment (iamdanfox) [inline gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/CreateManifestTask.java:137]: Would like to try and avoid making any network calls at configuration time if poss

File: gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/CreateManifestTask.java [hunk 2]
```diff
@@ -402,7 +471,7 @@ private void validateProjectVersion() {
                     task.getProductType().set(ext.getProductType());
                     task.setManifestFile(new File(project.getBuildDir(), "/deployment/manifest.yml"));
                     task.getProductDependencies().set(ext.getProductDependencies());
-                    task.getProductDependenciesConfig().set(project.provider(ext::getProductDependenciesConfig));
+                    task.setConfiguration(project.provider(ext::getProductDependenciesConfig));
```
Comment (iamdanfox) [inline gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/CreateManifestTask.java:474]: maybe for a follow-up - store a property<configuration> in the extension instead

**Comment by iamdanfox** [comment]
:+1:

File: gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/CreateManifestTask.java (hunk 1 above)
Comment (dansanduleac) [reply gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/CreateManifestTask.java:137]: this won't happen at configuration time

**Review by dansanduleac — APPROVED** [review]

**Pull request merged**